presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -118,6 +118,14 @@ public int positionOf(Block block, int position)
         return blockPositionByHash.get(getHashPositionOfElement(block, position));
     }
 
+    public Block getBlock()
+    {
+        if (containsNullElement) {
+            elementBlock.appendNull();
+        }
+        return elementBlock.build();
```

Comment:
can i call `TypedSet.getBlock()` more than once?

Reply:
Note once you called `getBlock`, the `TypedSet` becomes immutable.

That's why I am learning towards to generalize `JsonUtil#HashTable` for this common usage (i.e. constructing a Set over a block without copying the data), and deprecate `TypedSet` in the long term. 

We can discuss in person once @haozhun  is back.

Reply:
I opened an issue to discuss the solutions: https://github.com/prestodb/presto/issues/12033

Reply:
Right now it can't. But after TypedSet changed to use isDistinctFrom then we don't need to handle null differently, then this method can be called multiple times. Also all use cases of this method should only need to call it once, if needed.